internal/db/secrets.go

```diff
@@ -0,0 +1,262 @@
+package db
+
+import (
+	"context"
+	"database/sql"
+
+	"github.com/keegancsmith/sqlf"
+	"github.com/sourcegraph/sourcegraph/cmd/frontend/types"
+	"github.com/sourcegraph/sourcegraph/internal/db/dbconn"
+)
+
+// This object provides access to the secrets table
+type secrets struct{}
+
+// Returned when we can't retrieve the specific crypt object that we need
+type SecretNotFoundError struct {
```

Comment:
If this error is meant to be used for comparison, and since it doesn't contain any field inside I think it will be more useful to use a simple sentinel error instead.

```suggestion
var ErrSecretNotFound = errors.New("failed to find matching secret")
```

That way it will be easier to compare:

```go
// before

s, err := sc.Get(ctx context.Context, id int32)
if _, ok := err.(*db.SecretNotFoundError); ok {
}

// after
s, err := sc.Get(ctx context.Context, id int32)
if err == db.ErrSecretNotFound {
}

```